flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/spargel/VertexCentricIteration.java

```diff
@@ -138,69 +146,46 @@ public void setInput(DataSet<Vertex<VertexKey, VertexValue>> inputData) {
 		if (this.initialVertices == null) {
```

Comment:
I find the new logic of this method a bit confusing. Why do you set some configuration parameters here (degrees and direction) and the rest in the helper methods?
As far as I understand, you have 2 cases that matter here: the degrees option is set or not. This defines whether you create a simple vertex or you augment it with degree information, right?